server/src/main/java/org/jboss/as/server/ServerService.java

```diff
@@ -544,9 +547,10 @@ public synchronized void start(StartContext context) throws StartException {
             } else {
                 executorService = new EnhancedQueueExecutor.Builder()
                     .setCorePoolSize(getCorePoolSize(forDomain))
-                    .setMaximumPoolSize(1024)
+                    .setMaximumPoolSize(getMaxPoolSize())
                     .setKeepAliveTime(20L, TimeUnit.SECONDS)
                     .setThreadFactory(threadFactory)
+                    .setMBeanName(ENHANCED_EXECUTOR_MBEAN_NAME)
```

Comment:
I was not sure why you add this: it allows to have a consistent name for the MBean instead of a autogenerated names with a counter.

Reply:
Yes, it facilitates the testing; we can do it because there is only one MBean per server JVM / managed server (domain mode) JVM